File: src/problems/1074.ts
// 36진수

import fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "/input.txt";
var input = fs.readFileSync(__dirname + filePath).toString();

const zet = (
  q: number,
  x: number,
  y: number,
  r: number,
  c: number,
  i: number
) => {

  
};

// r: 세로, c: 가로

export const solution = (stdinInput: string) => {
  const [n, r, c] = stdinInput.trim().split(/\s/).map(Number);

  let q = 0;
  if (2 ** n / 2 > r && 2 ** n / 2 <= c) {
    q = 1;
  } else if (2 ** n / 2 > r && 2 ** n / 2 > c) {
    q = 2;
  } else if (2 ** n / 2 <= r && 2 ** n / 2 > c) {
    q = 3;
  } else {
    q = 4;
  }

  zet(q, )

};

solution(input);

File: src/problems/1074.py
N, Y, X = map(int, input().split())


def func(y, x, value, k):
    """
    value = 2사분면 좌상단 수
    l = 가로,세로의 절반
    q = 사분면
    degree = 각 사분면의 단위(좌상단 수)
    """

    l = 2 ** (k - 1)
    if x < l and y < l:
        y, x = y, x
        q = 0
    elif x >= l and y < l:
        y, x = y, x - l
        q = 1
    elif x < l and y >= l:
        y, x = y - l, x
        q = 2
    else:
        y, x = y - l, x - l
        q = 3

    degree = 2 ** (2 * (k - 1))
    value = value + degree * q

    if k == 0:
        print(int(value))
        return

    func(y, x, value, k - 1)


func(Y, X, 0, N)
